lib/Controller/ChatController.php

```diff
@@ -0,0 +1,192 @@
+<?php
+
+/**
+ *
+ * @copyright Copyright (c) 2017, Daniel Calviño Sánchez (danxuliu@gmail.com)
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\Spreed\Controller;
+
+use OCA\Spreed\Chat\ChatManager;
+use OCA\Spreed\Exceptions\RoomNotFoundException;
+use OCA\Spreed\Manager;
+use OCP\AppFramework\Http;
+use OCP\AppFramework\Http\DataResponse;
+use OCP\AppFramework\OCSController;
+use OCP\IRequest;
+use OCP\ISession;
+use OCP\IUserManager;
+
+class ChatController extends OCSController {
+
+	/** @var string */
+	private $userId;
+
+	/** @var IUserManager */
+	private $userManager;
+
+	/** @var ISession */
+	private $session;
+
+	/** @var Manager */
+	private $manager;
+
+	/** @var ChatManager */
+	private $chatManager;
+
+	/**
+	 * @param string $appName
+	 * @param string $UserId
+	 * @param IRequest $request
+	 * @param IUserManager $userManager
+	 * @param ISession $session
+	 * @param Manager $manager
+	 * @param ChatManager $chatManager
+	 */
+	public function __construct($appName,
+								$UserId,
+								IRequest $request,
+								IUserManager $userManager,
+								ISession $session,
+								Manager $manager,
+								ChatManager $chatManager) {
+		parent::__construct($appName, $request);
+
+		$this->userId = $UserId;
+		$this->userManager = $userManager;
+		$this->session = $session;
+		$this->manager = $manager;
+		$this->chatManager = $chatManager;
+	}
+
+	/**
+	 * @PublicPage
+	 *
+	 * Sends a new chat message to the given room.
+	 *
+	 * The author and timestamp are automatically set to the current user/guest
+	 * and time.
+	 *
+	 * @param string $token the room token
+	 * @param string $message the message to send
+	 * @return DataResponse the status code is "201 Created" if successful, and
+	 *         "404 Not found" if the room or session for a guest user was not
+	 *         found".
+	 */
+	public function sendMessage($token, $message) {
+		try {
+			$room = $this->manager->getRoomForParticipantByToken($token, $this->userId);
+		} catch (RoomNotFoundException $exception) {
+			return new DataResponse([], Http::STATUS_NOT_FOUND);
+		}
+
+		if ($this->userId === null) {
+			$actorType = 'guests';
+			$actorId = $this->session->get('spreed-session');
```

Comment:
I'm pretty sure this is too long?
Our session is 255, the actor id is only 64

Reply:
Oops, yes, you are totally right :-S

How could a guest be identified then? Maybe storing a 64 char long hash of the session? Any other (probably better :-P ) idea?

Reply:
Either that, or use a hard-coded string instead. But I don't know what this implies? Can guests edit all users messages then?
Well they can't anyway in our chat, right?

Reply:
The problem with a hard-coded string is that then there would be no way to know which guest sent which message; that is why I was using the session in first place.

Regarding the edition of messages no, they can not be edited in our chat interface. Although now that you mention it maybe they can be edited through the Comments DAV or something like that... I will have to check it.